iceoryx_utils/include/iceoryx_utils/internal/units/duration.hpp

```diff
@@ -186,49 +265,59 @@ class Duration
-    friend constexpr Duration duration_literals::operator"" _ns(long double);
     friend constexpr Duration duration_literals::operator"" _ns(unsigned long long int); // PRQA S 48
-    friend constexpr Duration duration_literals::operator"" _us(long double);
     friend constexpr Duration duration_literals::operator"" _us(unsigned long long int); // PRQA S 48
-    friend constexpr Duration duration_literals::operator"" _ms(long double);
     friend constexpr Duration duration_literals::operator"" _ms(unsigned long long int); // PRQA S 48
-    friend constexpr Duration duration_literals::operator"" _s(long double);
-    friend constexpr Duration duration_literals::operator"" _s(unsigned long long int); // PRQA S 48
-    friend constexpr Duration duration_literals::operator"" _m(long double);
-    friend constexpr Duration duration_literals::operator"" _m(unsigned long long int); // PRQA S 48
-    friend constexpr Duration duration_literals::operator"" _h(long double);
-    friend constexpr Duration duration_literals::operator"" _h(unsigned long long int); // PRQA S 48
-    friend constexpr Duration duration_literals::operator"" _d(long double);
-    friend constexpr Duration duration_literals::operator"" _d(unsigned long long int); // PRQA S 48
+    friend constexpr Duration duration_literals::operator"" _s(unsigned long long int);  // PRQA S 48
+    friend constexpr Duration duration_literals::operator"" _m(unsigned long long int);  // PRQA S 48
+    friend constexpr Duration duration_literals::operator"" _h(unsigned long long int);  // PRQA S 48
+    friend constexpr Duration duration_literals::operator"" _d(unsigned long long int);  // PRQA S 48
+
+    template <typename T>
+    friend constexpr Duration operator*(const T& left, const Duration& right);
+
+    friend std::ostream& operator<<(std::ostream& stream, const Duration& t);
 
   private:
-    /// @brief constructor needs to be private to ensure a unit safe usage of duration
-    constexpr explicit Duration(const long double durationInSeconds);
-    long double durationInSeconds{0.0};
+    template <typename T>
+    inline constexpr Duration
+    multiplySeconds(const uint64_t seconds, const std::enable_if_t<!std::is_floating_point<T>::value, T>& right) const;
+    template <typename T>
+    inline constexpr Duration multiplySeconds(const uint64_t seconds,
+                                              const std::enable_if_t<std::is_floating_point<T>::value, T>& right) const;
+    template <typename T>
+    inline constexpr Duration
+    multiplyNanoseconds(const uint32_t nanoseconds,
+                        const std::enable_if_t<!std::is_floating_point<T>::value, T>& right) const;
+    template <typename T>
+    inline constexpr Duration
+    multiplyNanoseconds(const uint32_t nanoseconds,
+                        const std::enable_if_t<std::is_floating_point<T>::value, T>& right) const;
+
+  private:
+    uint64_t m_seconds{0};
```

Comment:
Are U correctly initialized? ;-)

Reply:
yes, it's initialized with 64 bits set to 0 😉

Reply:
btw, what do you think of
```c++
#define 0 0U
```
or
```c++
constexpr unsigned U{0U};
```